src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -78,9 +79,9 @@ private static KNNVectorFieldMapper toType(FieldMapper in) {
                 m -> toType(m).stored, false);
         private final Parameter<Boolean> hasDocValues = Parameter.boolParam("doc_values", false,
                 m -> toType(m).hasDocValues,  true);
-        private final Parameter<Integer> dimension = new Parameter<>(KNNConstants.DIMENSION, false, -1,
-                (n, o) -> {
-                            int value = XContentMapValues.nodeIntegerValue(o);
+        private final Parameter<Integer> dimension = new Parameter<>(KNNConstants.DIMENSION, false, () -> -1,
+                (n, c, o) -> {
+                            int value = (o == null ? null : XContentMapValues.nodeIntegerValue(o));
```

Comment:
if o is null, then i don't think we can assign null to value since it is primitive.

Reply:
Good catch. Fixed.